tidb-binlog/pump_client/client.go

```diff
@@ -321,42 +342,42 @@ func (c *PumpsClient) WriteBinlog(binlog *pb.Binlog) error {
 	log.Info("[pumps client] write binlog to avaliable pumps all failed, will try unavaliable pumps")
 	pump, err1 := c.backoffWriteBinlog(req, binlog.Tp, binlog.StartTs)
 	if err1 == nil {
+		choosePump = pump
 		return nil
 	}
-	choosePump = pump
 
 	return errors.Errorf("write binlog failed, the last error %v", err)
 }
 
+// return directly for non p-binlog
+// try every online pump for p-binlog
 func (c *PumpsClient) backoffWriteBinlog(req *pb.WriteBinlogReq, binlogType pb.BinlogType, startTS int64) (pump *PumpStatus, err error) {
 	if binlogType != pb.BinlogType_Prewrite {
 		// never return error for commit/rollback binlog.
 		return nil, nil
 	}
 
-	unAvaliablePumps := make([]*PumpStatus, 0, 3)
+	allPumps := make([]*PumpStatus, 0, 3)
 	c.RLock()
-	for _, pump := range c.Pumps.UnAvaliablePumps {
-		unAvaliablePumps = append(unAvaliablePumps, pump)
+	for _, pump := range c.Pumps.Pumps {
+		allPumps = append(allPumps, pump)
 	}
 	c.RUnlock()
 
 	var resp *pb.WriteBinlogResp
 	// send binlog to unavaliable pumps to retry again.
-	for _, pump := range unAvaliablePumps {
-		if !pump.IsUsable() {
+	for _, pump := range allPumps {
+		if !pump.ShouldBeUsable() {
 			continue
 		}
 
-		pump.ResetGrpcClient()
-
 		resp, err = pump.WriteBinlog(req, c.BinlogWriteTimeout)
 		if err == nil {
 			if resp.Errmsg != "" {
 				err = errors.New(resp.Errmsg)
 			} else {
 				// if this pump can write binlog success, set this pump to avaliable.
-				log.Debug("[pumps client] write binlog to unavaliable pump success, set this pump to avaliable", zap.String("NodeID", pump.NodeID))
+				log.Info("[pumps client] write binlog to pump success, set this pump to avaliable", zap.String("NodeID", pump.NodeID))
 				c.setPumpAvaliable(pump, true)
```

Comment:
What's the behavior of `setPumpAvailable` when `pump` is already available?

Reply:
it takes no effect